Source/Debugging/Generators/Documents/DebuggingHandlerDocumentGenerator.cs

```diff
@@ -0,0 +1,123 @@
+
+namespace Dolittle.AspNetCore.Generators.Documents
+{
+    /// <summary>
+    /// An implementation of <see cref="IDebuggingHandlerDocumentGenerator"/>.
+    /// </summary>
+    public class DebuggingHandlerDocumentGenerator : IDebuggingHandlerDocumentGenerator
+    {
+        readonly ISchemaGenerator _schemaGenerator;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="DebuggingHandlerDocumentGenerator"/> class.
+        /// </summary>
+        /// <param name="schemaGenerator">The <see cref="ISchemaGenerator"/> used to generate schemas.</param>
+        public DebuggingHandlerDocumentGenerator(ISchemaGenerator schemaGenerator)
+        {
+            _schemaGenerator = schemaGenerator;
+        }
+
+        /// <inheritdoc/>
+        public OpenApiDocument GenerateFor(IDebuggingHandler handler)
+        {
+            var repository = new SchemaRepository();
+            var paths = new OpenApiPaths();
+            GeneratePaths(paths, handler, repository);
+
+            return new OpenApiDocument
+            {
+                Info = new OpenApiInfo
+                {
+                    Title = handler.Title,
+                },
+                Paths = paths,
+                Components = new OpenApiComponents
+                {
+                    Schemas = repository.Schemas,
+                },
+            };
+        }
+
+        void GeneratePaths(OpenApiPaths paths, IDebuggingHandler handler, SchemaRepository repository)
+        {
+            foreach ((var path, var artifact) in handler.Artifacts)
+            {
+                var item = new OpenApiPathItem();
+
+                AddGetOperation(handler, item, GenerateOperation(handler, artifact, repository));
+                AddPostOperation(handler, item, GenerateOperation(handler, artifact, repository));
+
+                var tag = path.ToString().Contains('/', StringComparison.InvariantCultureIgnoreCase) ? path.ToString().Split('/')[1] : path.ToString();
+                foreach ((_, var operation) in item.Operations)
+                {
+                    operation.Tags = new[] { new OpenApiTag { Name = tag } };
+                }
+
+                paths.Add(path, item);
+            }
+        }
+
+        void AddGetOperation(IDebuggingHandler handler, OpenApiPathItem item, OpenApiOperation operation)
+        {
+            if (handler.GetType().ImplementsOpenGeneric(typeof(ICanHandleGetRequests<>)))
+            {
+                item.AddOperation(OperationType.Get, operation);
+            }
+        }
+
+        void AddPostOperation(IDebuggingHandler handler, OpenApiPathItem item, OpenApiOperation operation)
+        {
+            if (handler.GetType().ImplementsOpenGeneric(typeof(ICanHandlePostRequests<>)))
+            {
+                item.AddOperation(OperationType.Post, operation);
+            }
+        }
+
+        // Generates the responses and the default JSON RequestBody (if applicable) for an operation.
```

Comment:
Probably not needed, the method name is pretty clear. Potentially the method name could be `GenerateApiOperation` - but I think its within the context it is quite explanatory on its own. JSON is implicit in the context as well.

Comments within methods or for explaining private methods tend to be a code-smell and often at times reflect lack of good naming, which I don't believe is the case here.

```suggestion
```